cppcache/src/PdxRemoteWriter.hpp

```diff
@@ -45,10 +47,10 @@ class PdxRemoteWriter : public PdxLocalWriter {
   PdxRemoteWriter() = default;
 
   PdxRemoteWriter(DataOutput& output, std::shared_ptr<PdxType> pdxType,
-                  std::shared_ptr<PdxRemotePreservedData> preservedData,
                   std::shared_ptr<PdxTypeRegistry> pdxTypeRegistry);
 
-  PdxRemoteWriter(DataOutput& output, std::string pdxClassName,
+  PdxRemoteWriter(DataOutput& output, std::shared_ptr<PdxType> pdxType,
+                  std::shared_ptr<PdxRemotePreservedData> preservedData,
```

Comment:
It this really feels like an extension of the other constructor with the optional `preservedData` parameters to maybe re-order the parameters to look like that.